clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -626,12 +704,14 @@ public synchronized void maybeTransitionToErrorState(RuntimeException exception)
             if (canBumpEpoch() && !isCompleting()) {
                 epochBumpRequired = true;
             }
-            transitionToAbortableError(exception);
+            transitionToAbortableError(exception, invalidStateDetectionStrategy);
         }
     }
-
-    synchronized void handleFailedBatch(ProducerBatch batch, RuntimeException exception, boolean adjustSequenceNumbers) {
-        maybeTransitionToErrorState(exception);
+    synchronized void handleFailedBatch(ProducerBatch batch,
```

Comment:
is this ever called with InvalidStateDetectionStrategy.FOREGROUND?
if not, we might not need the extra parameter, and always use BACKGROUND

Reply:
@urbandan There is only one place that `handleFailedBatch` is called with `FOREGROUND`: the `TransactionManagerTest.testForegroundInvalidStateTransitionIsRecoverable()` method.

That unit test was calling `handleFailedBatch` as a simple way to simulate the state transition error. I updated the test using a different means. As such, I was able to remove that parameter.

Thanks!